presto-main/src/test/java/com/facebook/presto/execution/buffer/TestSpoolingOutputBuffer.java

```diff
@@ -0,0 +1,541 @@
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.DataSize;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.execution.buffer.BufferResult.emptyResults;
+import static com.facebook.presto.execution.buffer.BufferState.OPEN;
+import static com.facebook.presto.execution.buffer.BufferState.TERMINAL_BUFFER_STATES;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.MAX_WAIT;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.NO_WAIT;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.acknowledgeBufferResult;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.addPage;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.addPages;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.assertBufferResultEquals;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.createBufferResult;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.createPage;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.getBufferResult;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.getFuture;
+import static com.facebook.presto.execution.buffer.BufferTestUtils.sizeOfPages;
+import static com.facebook.presto.execution.buffer.OutputBuffers.BufferType.PARTITIONED;
+import static com.facebook.presto.execution.buffer.OutputBuffers.BufferType.SPOOLING;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestSpoolingOutputBuffer
+{
+    private static final String TASK_INSTANCE_ID = "task-instance-id";
+
+    private static final List<BigintType> TYPES = ImmutableList.of(BIGINT);
+    private static final OutputBuffers.OutputBufferId BUFFER_ID = new OutputBuffers.OutputBufferId(0);
+    private static final OutputBuffers.OutputBufferId INVALID_BUFFER_ID = new OutputBuffers.OutputBufferId(1);
+
+    private static final QueryIdGenerator queryIdGenerator = new QueryIdGenerator();
+    private static final List<Path> spilledPaths = new LinkedList<>();
+
+    private ScheduledExecutorService stateNotificationExecutor;
+
+    @BeforeClass
+    public void setUp()
+    {
+        File tempDir = Files.createTempDir();
+        spilledPaths.add(tempDir.toPath());
+        stateNotificationExecutor = newScheduledThreadPool(5, daemonThreadsNamed("test-%s"));
```

Comment:
Maybe make "test-%s" as a static variable, as a good practice.
Also 5 can be a variable outside as well with a good describing name